File: 0000-0999/0019 - Degree of Symmetry - Simmetriya dərəcəsi - Степень симметрии.py
# Input X as intenger
x =int(input())
# Counter
say = 0
y = 10

# Find the length of X
xLen = len(str(x))

# We have 2 option there | Are xLen even number or odd number
if xLen%2 == 0:   # If xLen even number
  for i in range(xLen//2):           # We need to use // for divide, because of we need int, dont need float number !
    if int(str(x)[i]) == int((x%10)/(y//10)):
      say+=1
    x/=10
    
# We have a small requirement for odd numbers. | Input: 12321  Result: 3
else:   # If xLen odd number
  for i in range(xLen//2): 
    if int(str(x)[i]) == int((x%10)/(y//10)):
      say+=1
    x/=10
    # Same loop as even numbers, but add 1 at the end for the correct answer !
  say+=1
    
print(say)

File: 0000-0999/0019 - Degree of Symmetry - Simmetriya dərəcəsi - Степень симметрии.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int x,xCopy,xCopy2,sum=0,i=0,xLen=0,z,uzunluq;
	// Input X
	cin>>x;
	// Create copy of X
	xCopy=x;
	xCopy2=x;
	// Find the length of X
	do {
     xLen++; 
     xCopy /= 10;
} while (xCopy);
	// The length of X = xLen 
	uzunluq=xLen;
	
	// We need to check 2 option here | Are X odd number or even number
	 if(xLen%2 == 0){ // If X even number
		for(i;i<xLen/2;i++){
			z=pow(10, uzunluq-1);
			if((x/z)%10 == xCopy2%10){
				sum++;
			}
			uzunluq--;
			xCopy2/=10;
		}
	// We have a small requirement for odd numbers. | Input: 12321  Result: 3
	} 
	else{ // If X odd number
		for(i;i<xLen/2;i++){
			z=pow(10, uzunluq-1);
			if((x/z)%10 == xCopy2%10){
				sum++;
			}
			uzunluq--;
			xCopy2/=10;
		}
		// Same loop as even numbers, but add 1 at the end for the correct answer !
		sum++;
	}
	
	cout<<sum;
}
